pep-0484.txt

```diff
@@ -1294,6 +1296,11 @@ collections (e.g. ``List``), types representing generic
 collection ABCs (e.g. ``Sequence``), and a small collection of
 convenience definitions.
 
+Note that special type constructs, such as ``Any``, ``TypeVar``,
+``Union``, and ``Generic`` are only supported in the type annotation
+context and will raise ``TypeError`` if appear in ``isinstance``
+or ``issubclass``.
```

Comment:
Actually TypeVar is a very different construct, though maybe type variables defined *using* TypeVar are indeed such special constructs. Also, Generic is not supported in a type annotation context, it may only be used as a base class.